static/app/utils/replayCount/replayCountCache.tsx

```diff
@@ -0,0 +1,136 @@
+import {createContext, ReactNode, useCallback, useContext} from 'react';
+
+import {ApiResult} from 'sentry/api';
+import {Organization} from 'sentry/types';
+import {ApiQueryKey} from 'sentry/utils/queryClient';
+import useBufferedQuery from 'sentry/utils/replayCount/useBufferedQuery';
+
+interface QueryKeyGenProps {
+  dataSource: string;
+  fieldName: string;
+  organization: Organization;
+}
+
+interface Props {
+  children: ReactNode;
+  queryKeyGenProps: QueryKeyGenProps;
+}
+
+type CountValue = undefined | number;
+type ExistsValue = undefined | boolean;
+type CountState = Record<string, CountValue>;
+type ExistsState = Record<string, ExistsValue>;
+
+interface TContext {
+  getMany: (ids: readonly string[]) => CountState;
+  getOne: (id: string) => CountValue;
+  hasMany: (ids: readonly string[]) => ExistsState;
+  hasOne: (id: string) => ExistsValue;
+}
+
+const DEFAULT_CONTEXT: TContext = {
+  getMany: () => ({}),
+  getOne: () => undefined,
+  hasMany: () => ({}),
+  hasOne: () => undefined,
+};
+
+const ReplayCacheCountContext = createContext<TContext>(DEFAULT_CONTEXT);
+
+function queryKeyGen({dataSource, fieldName, organization}: QueryKeyGenProps) {
+  return (ids: readonly string[]): ApiQueryKey => [
+    `/organizations/${organization.slug}/replay-count/`,
+    {
+      query: {
+        data_source: dataSource,
+        project: -1,
+        statsPeriod: '14d',
```

Comment:
this needs to be dynamic, for https://github.com/getsentry/team-replay/issues/318